### PR TITLE
Added logging changing model properties. Closed #8

### LINKED ISSUE #8: Add logging when changing model properties

### PR DESCRIPTION
**Pull request by artemqdh**

### CODE REVIEW

File: Program/Program.cs
Comment (TopAcademy2022) [inline Program/Program.cs:1]: Remove this file.

File: Program/Program.csproj
Comment (TopAcademy2022) [inline Program/Program.csproj:1]: Remove this file.

File: music-store.sln
Comment (TopAcademy2022) [inline music-store.sln:1]: Reverse changes in file.

File: music-store/Models/Entities/MusicBand.cs
Comment (TopAcademy2022) [inline music-store/Models/Entities/MusicBand.cs:1]: Reverse changes in file.

File: music-store/Models/Entities/VinylRecord.cs [hunk 1]
```diff
@@ -1,25 +1,138 @@
 ﻿using System;
+using System.IO;
 
 namespace music_store.Models.Entities
 {
 	public class VinylRecord
 	{
+		private static readonly string LogFilePath = "Logs/changes.log";
```
Comment (TopAcademy2022) [inline music-store/Models/Entities/VinylRecord.cs:8]: Relocate this line.

File: music-store/Models/Entities/VinylRecord.cs
Comment (TopAcademy2022) [inline music-store/Models/Entities/VinylRecord.cs:1]: Not created private fields.

File: music-store/Models/Entities/VinylRecord.cs [hunk 2]
```diff
@@ -1,25 +1,138 @@
...
+		}
 
-		public uint CostPrice { get; set; }
+		/*!
+		 *	@brief Saves the changes to file
+		 *	param[in] propertyName - Name of the column
+		 *	param[in] oldValue - old value of column
+		 *	param[in] newValue - new value of column
+		 */
+		private static void LogChange(string propertyName, string oldValue, string newValue)
```
Comment (TopAcademy2022) [inline music-store/Models/Entities/VinylRecord.cs:130]: Using ILogger.

File: music-store/Models/Entities/MusicGenre.cs
Comment (TopAcademy2022) [inline music-store/Models/Entities/MusicGenre.cs:1]: Reverse changes in file.

File: music-store/Models/Entities/MusicPublisher.cs
Comment (TopAcademy2022) [inline music-store/Models/Entities/MusicPublisher.cs:1]: Reverse changes in file.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]
Use type T class.

File: Program/Program.cs
Comment (artemqdh) [inline Program/Program.cs:1]: delete program.cs? and program.csproj?

Comment (TopAcademy2022) [reply Program/Program.cs:1]: Yes.

File: music-store/Models/Entities/VinylRecord.cs [hunk 3]
```diff
@@ -1,25 +1,138 @@
 ﻿using System;
+using music_store.Services;
 
 namespace music_store.Models.Entities
 {
 	public class VinylRecord
 	{
+		private readonly Logger? _logger;
```
Comment (TopAcademy2022) [inline music-store/Models/Entities/VinylRecord.cs:8]: Move in another class. Create base class for models.

File: music-store/Models/Entities/VinylRecord.cs [hunk 4]
```diff
@@ -1,25 +1,138 @@
...
+
+		public VinylRecord(Logger? logger = null)
```
Comment (TopAcademy2022) [inline music-store/Models/Entities/VinylRecord.cs:10]: See above.

File: music-store/Models/Entities/VinylRecord.cs [hunk 5]
```diff
@@ -1,25 +1,138 @@
...
+		{
+			_logger = logger;
+		}
+
 		public int Id { get; set; }
 
-		public string Name { get; set; } = null!;
+		private string _name = null!;
```
Comment (TopAcademy2022) [inline music-store/Models/Entities/VinylRecord.cs:17]: Don't create fields.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

File: Program/Program.cs
Comment (TopAcademy2022) [inline Program/Program.cs:1]: Revome file.

File: music-store.sln
Comment (TopAcademy2022) [inline music-store.sln:1]: Request chages.

File: music-store/Models/Entities/MusicBand.cs
Comment (TopAcademy2022) [inline music-store/Models/Entities/MusicBand.cs:1]: Remove spaces.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

**Review by TopAcademy2022 — DISMISSED** [review]

**Pull request closed without merging**